internal/ansible/ansible.go

```diff
@@ -20,131 +20,260 @@ import (
+	if err != nil {
+		return nil, err
+	}
 
-	// Read playbooks filename from dir
-	pbList, err := readDir(p.Dir, p.ExcludedFiles)
+	wd, err := os.Getwd()
 	if err != nil {
 		return nil, err
 	}
-	return NewAnsiblePlaybook(WithPlaybooks(pbList),
-		// `ansible-playbook` cmd output JSON Serialization
-		WithStdoutCallback("json"),
-		// e.g options.AnsibleConnectionOptions{Connection: "local"}
-		WithConnectionOptions(&options.AnsibleConnectionOptions{}),
-		// e.g playbook.AnsiblePlaybookOptions{Inventory: "127.0.0.1,"}
-		WithOptions(&playbook.AnsiblePlaybookOptions{})), nil
+
+	return func(verbosity int) *exec.Cmd {
+		// check the verbosity since the exec.Command will fail if an arg as "" or " " be informed
+
+		cmdOptions := []string{
+			"--role", roleName,
+			"--roles-path", rolePath,
+			"--hosts", "localhost",
+		}
+		cmdArgs := []string{"run", wd}
+
+		if verbosity > 0 {
+			cmdOptions = append(cmdOptions, runnerutil.AnsibleVerbosityString(verbosity))
+		}
+		ansibleGathering := os.Getenv("ANSIBLE_GATHERING")
+
+		// When running a role directly, ansible-runner does not respect the ANSIBLE_GATHERING
+		// environment variable, so we need to skip fact collection manually
+		if ansibleGathering == "explicit" {
+			cmdOptions = append(cmdOptions, "--role-skip-facts")
+		}
+
+		// gosec is disabled here because of G204. We should pay attention that user can't
+		// make command injection via command argument
+		return exec.Command(runnerBinary, append(cmdArgs, cmdOptions...)...) //nolint:gosec
+	}, nil
+}
+
+// galaxyInstall Install non-exists collections with ansible-galaxy cli
+func (p Parameters) galaxyInstall() error {
+	galaxyBinary, err := galaxyutil.GalaxyBinary()
+	if err != nil {
+		return err
+	}
+
+	requirementsFilePath := runnerutil.GetFullPath(p.WorkingDir, galaxyutil.RequirementsFile)
+
+	cmdArgs := []string{"collection", "install"}
+	cmdOptions := []string{
+		"--requirements-file", requirementsFilePath,
+	}
+
+	// ansible-galaxy is by default verbose
+	cmdOptions = append(cmdOptions, "--verbose")
+
+	// gosec is disabled here because of G204. We should pay attention that user can't
+	// make command injection via command argument
+	dc := exec.Command(galaxyBinary, append(cmdArgs, cmdOptions...)...) //nolint:gosec
+
+	out, err := dc.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to install galaxy collections: %v: %v", string(out), err)
+	}
+	return nil
 }
 
-// NewAnsiblePlaybook returns a pbCmd that will be used as ansible-playbook client
-func NewAnsiblePlaybook(o ...PlaybookOption) *PbCmd {
+// Init initializes a new runner from parameters
+func (p Parameters) Init(ctx context.Context, cr *v1alpha1.AnsibleRun) (*Runner, error) {
+	if err := p.galaxyInstall(); err != nil {
+		return nil, err
+	}
+
+	addRolePlaybookPaths(p, cr)
 
-	pb := &playbook.AnsiblePlaybookCmd{
-		Playbooks: []string{},
+	if cr.Spec.ForProvider.Source == v1alpha1.ConfigurationSourceRemote {
```

Comment:
Should it be `cr.Spec.ForProvider.Source == 1alpha1. ConfigurationSourceInline` since it is about to override the user input values using the pre-defined playbook.yml file.

Also, I'm suspecting the `role.yml` file is not easy to be applicable in this case as it usually includes a set of yaml files w/ well organized directory structure. Maybe we can just focus on `playbook.yml` for the inline mode?

Reply:
👍 I fix this 
Yep for inline mode online `playbook.yml` , as you said above the inline mode should be quick win that help users to make simple test or to append complex remote ansible project :)